packages/x-scheduler/src/primitives/use-event-calendar/EventCalendarStore.ts

```diff
@@ -325,6 +332,40 @@ export class EventCalendarStore extends Store<State> {
     onEventsChange?.(updatedEvents);
   };
 
+  /**
+   * Applies the data from the placeholder occurrence to the event it represents.
+   */
+  public async applyOccurrencePlaceholder(
+    data: CalendarOccurrencePlaceholder,
+    chooseRecurringEventScope?: () => Promise<RecurringUpdateEventScope>,
```

Comment:
@rita-codes my idea here is that the UI can pass a promise that will resolve when the scope has been decided.
But if this approach is too clunky on the styled layer, feel free to modify this method / replace it with a hook etc...

I alsolet the event creation part empty for your PR 👍